### PR TITLE
fix(public-cloud): fix object storage high perf wrongly listed

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | DTRSD-75508 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I fixed object storage high perf wrongly listed.

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/notebooks/add/components/notebook-attach/notebook-attach.controller.js [hunk 1]
```diff
@@ -96,19 +96,17 @@ export default class NotebookAttachController {
...
-      .map(({ name, region }) => {
-        return {
-          name,
-          region,
-          description: `${name} - ${region}`,
-        };
+      .flatMap(({ name, region, isHighPerfStorage }) => {
+        return !isHighPerfStorage
+          ? { name, region, description: `${name} - ${region}` }
+          : null;
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/notebooks/add/components/notebook-attach/notebook-attach.controller.js:109]: Don't you need to filter those null values after?

Comment (ghyenne) [reply packages/manager/modules/pci/src/projects/project/notebooks/add/components/notebook-attach/notebook-attach.controller.js:109]: With the flatMap the null values do not go back into the table.

Comment (ghyenne) [reply packages/manager/modules/pci/src/projects/project/notebooks/add/components/notebook-attach/notebook-attach.controller.js]: Indeed I made a mistake, it is [ ] and not null that I need.
I have applied the change.

**Review by frenautvh — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/notebooks/add/components/notebook-attach/notebook-attach.controller.js [hunk 2]
```diff
@@ -96,19 +96,17 @@ export default class NotebookAttachController {
 
   filterStorages() {
     this.filteredStorages = this.storages
-      // Remove containers that are already on volume list
+      // Remove containers that are already on volume list and and which are isHighPerfStorage.
       .filter(({ name, region }) => {
         return !this.notebookModel.volumes
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/pci/src/projects/project/notebooks/add/components/notebook-attach/notebook-attach.controller.js]: Indeed, I think I searched much to hard when there were more simple solutions.

**Review by frenautvh — DISMISSED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by radireddy — APPROVED** [review]

**Review by vks-ovh — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**